data/lib/src/datasource/shared_space_document_datasource.dart

```diff
@@ -39,4 +39,7 @@ abstract class SharedSpaceDocumentDataSource {
       Uri baseUrl,
       SharedSpaceId sharedSpaceId,
       {WorkGroupNodeId parentNodeId});
+
+  Future<List<WorkGroupNode>> getAllChildNodes(
+    SharedSpaceId sharedSpaceId, [WorkGroupNodeId parentNodeId]);
```

Comment:
what is `[WorkGroupNodeId parentNodeId]`?, it is an optional parameter?

Reply:
please also reformat code

Reply:
@hoangdat : Yeah, in case we need to get files at root workgroup, just provide `sharedSpaceId`

Reply:
but as I know, [] is for list, not for optional parameter :(

Reply:
It's [Optional positional parameters](https://dart.dev/guides/language/language-tour#parameters). It's also a type of optional parameter. But sorry, I think I should use named parameter `{ }`. I will fix it

Reply:
please consider carefully, `{}` use for default parameters too, if not, null is default.